hledger-mode.el

```diff
@@ -58,9 +58,25 @@
   :type 'face
   :group 'hledger)
 
+(defcustom hledger-refresh-completions-idle-delay 1
+  "Update completions in file when Emacs has been idle for this many seconds.")
+
+(defcustom hledger-invalidate-completions '(on-idle)
```

Comment:
I am trying to understand what using `on-idle` as the default choice means here. If Emacs is idle for some time, it doesn't necessarily signal in any way that the accounts list has been modified and needs an update. But then I do not have a better suggestion here. 

If we could figure out that the user entered a new account or had to fallback to typing instead of picking up an item from possible completions, then that could be a great default here. 

The problem I have with this default value is that I rarely update/add new accounts to my hledger file, maybe once in a few months and sometimes not even during a whole year. With this default, every second that Emacs is idle, it's setting this flag to update the list of accounts.

Reply:
I think we should probably use a larger value for `hledger-refresh-completions-idle-delay` for people like me who rarely update the list of accounts they use.

Reply:
> If we could figure out that the user entered a new account or had to fallback to typing instead of picking up an item from possible completions, then that could be a great default here.

Yeah, the other hooks are really just a proxy for that.

> The problem I have with this default value is that I rarely update/add new accounts to my hledger file, maybe once in a few months and sometimes not even during a whole year. With this default, every second that Emacs is idle, it's setting this flag to update the list of accounts.

I see what you mean. One second was a pretty arbitrary choice, mostly so that I could see it working; I think you’re absolutely right that it’s not necessary. I was about to suggest 10 minutes, but I’m starting to wonder if it’s just not worth updating on idle, because at that point it doesn’t have a lot of value.

Should I just drop the `on-idle` value?

Reply:
> Should I just drop the on-idle value in the default?

I think a better option would be something like `on-init`. Every time `hledger-mode` is enabled in a buffer, `hledger-accounts-cache` can get updated and would stay the same until either a new buffer with `hledger-mode` is opened or the same buffer is killed and created again.

Reply:
Right, that’s how it works at present, and I’ve preserved that behaviour in this code, so all the account refreshing code is an addition. I’ll drop `on-init` and clarify in the documentation that this variable doesn’t replace the normal updates done when opening a file.

Reply:
Cool 👍